src/mscorlib/src/Internal/Runtime/CompilerServices/Unsafe.cs

```diff
@@ -94,117 +107,249 @@ public static int SizeOf<T>()
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static ref T Add<T>(ref T source, int elementOffset)
         {
+#if PROJECTN
+            return ref AddByteOffset(ref source, (IntPtr)(elementOffset * (nint)SizeOf<T>()));
+#else
             // The body of this function will be replaced by the EE with unsafe code!!!
             // See getILIntrinsicImplementationForUnsafe for how this happens.
             typeof(T).ToString(); // Type token used by the actual method body
```

Comment:
These methods with `typeof(T).ToString()` in them should be rather ifdefed for CoreCLR, like:
```
#if CORECLR
    typeof(T).ToString();.....
#else
...
#endif
```